### DISCUSSION TITLE
Optional configurations based on environment variables

### DISCUSSION THREAD

**Discussion by Hate9** [ghostty-org/ghostty#4002] [Feature Requests, Ideas] [answered]
One feature I really appreciated with Kitty was the ability to use environment variables in my configuration. This meant that I could use different sub-config files based on what desktop environment I was running, which was *really nice* for making my terminal always match my rice.

Is there a way of doing this with Ghostty? And if not, does that seem like something that would be a good addition?

**Comment by jcollie** [comment]
That is not possible right now. As for adding it as a feature it would _really_ complicate the config file parsing so it's unlikely.

**Comment by mitchellh** [comment, marked as answer]
There was a somewhat recent duplicate where I gave a more detailed response to this.
